packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -34,64 +36,144 @@ export function Tabs<T extends object>(props: SpectrumTabsProps<T>) {
     density,
     ...otherProps
   } = props;
+
   let ref = useRef<HTMLDivElement>();
+  let wrapperRef = useRef<HTMLDivElement>();
   let state = useSingleSelectListState<T>({
     ...props,
     onSelectionChange
   });
 
+  let {direction} = useLocale();
   let {styleProps} = useStyleProps(otherProps);
   let {tabListProps, tabPanelProps} = useTabs(props, state, ref);
+  let [collapse, setCollapse] = useValueEffect(false);
   let [selectedTab, setSelectedTab] = useState<HTMLElement>();
 
+  let lastSelectedKey = useRef(state.selectedItem);
+  useEffect(() => {
+    // Ensure a tab is always selected (in case no selected key was specified or if selected item was deleted from collection)
+    if (state.selectionManager.isEmpty || !state.collection.getItem(state.selectedKey)) {
+      // if there was a prior key specified, try to set the next key as the selected key
+      // fall back to the previous key if next key doesn't exist
+      if (state.selectedKey != null) {
+        let nextKey = lastSelectedKey.current.nextKey;
+        let replacementKey = nextKey != null && state.collection.getItem(nextKey) ? nextKey : lastSelectedKey.current.prevKey;
+        replacementKey = state.collection.getItem(replacementKey) ? replacementKey : state.collection.getLastKey();
+        state.selectionManager.replaceSelection(replacementKey);
+      } else {
+        state.selectionManager.replaceSelection(state.collection.getFirstKey());
+      }
+    }
+
+    lastSelectedKey.current = state.selectedItem;
+  }, [state]);
```

Comment:
As per review comment @ktabors, updated the behavior when the selected tab is removed from the collection (. It now attempts to use the deleted tab's `nextKey` as the new selected key if it exists in the collection. If that key doesn't exist then it tries to use the deleted tab's `prevKey`, falling back to the collection's last key if that doesn't exist. This doesn't handle multiple tab deletions that well (i.e. it doesn't iterate upwards through all deleted keys), but committing it in this form so we can discuss what we believe the final behavior should be.

Other opinions welcome

Reply:
The above logic will default to making the  collection's `lastKey` the selected tab if the deleted key's `nextKey` and `prevKey` have both been deleted from the TabList

Reply:
and what if the last key is disabled?

Reply:
yeah, that is the problematic case. I was playing around with using the TabsKeyboardDelegate's `getNextKey`/`getPreviousKey` to search for valid non-disabled keys, but what if the entire collection of tabs is disabled? Gonna bring this up in standup tmrw